src/main/scala/com/sclasen/akka/kafka/AkkaDirectConsumer.scala

```diff
@@ -0,0 +1,164 @@
+package com.sclasen.akka.kafka
+
+import java.util.concurrent.{Executors, TimeUnit}
+
+import akka.actor._
+import akka.util.Timeout
+import kafka.consumer._
+import kafka.message.MessageAndMetadata
+import kafka.serializer.Decoder
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class AkkaDirectConsumer[Key,Msg](props:AkkaDirectConsumerProps[Key,Msg]) {
+
+  import AkkaConsumer._
+
+  lazy val connector = createConnection(props)
+  lazy val executor = Executors.newFixedThreadPool(props.streams)
+
+  def kafkaConsumerProps(zkConnect:String, groupId:String) = {
+    val consumerConfig = props.system.settings.config.getConfig("kafka.consumer")
+    val required = Set("zookeeper.connect" -> zkConnect, "group.id" -> groupId, "consumer.timeout.ms" -> "-1", "auto.commit.enable" -> "true")
+    val requiredKeys = required.map(_._1)
+    val consumerProps = consumerConfig.entrySet().asScala
+      .filter( k => !requiredKeys.contains(k.getKey) )
+      .map{
+        entry => entry.getKey -> consumerConfig.getString(entry.getKey)
+      } ++ required
+    toProps(consumerProps)
+  }
+
+  def kafkaConsumer(zkConnect:String, groupId:String) = {
+    Consumer.create(new ConsumerConfig(kafkaConsumerProps(zkConnect, groupId)))
+  }
+
+  def createConnection(props:AkkaDirectConsumerProps[Key,Msg]) =  {
+    import props._
+    val consumerConfig = new ConsumerConfig(kafkaConsumerProps(zkConnect, group))
+    Consumer.create(consumerConfig)
+  }
+
+  def createStream = {
+    props.topicFilterOrTopic match {
+      case Left(t) =>
+        props.system.log.info(s"createStream for topic: ${t}")
+        connector.createMessageStreamsByFilter(t, props.streams, props.keyDecoder, props.msgDecoder)
+      case Right(t) =>
+        props.system.log.info(s"createStream for topic: ${t}")
+        connector.createMessageStreams(Map(t -> props.streams), props.keyDecoder, props.msgDecoder).apply(t)
+    }
+  }
+
+  def start(): Future[Unit] = Future {
+    val streams = createStream
+    streams.foreach { stream =>
+      executor.submit(new Runnable() {
+        def run(): Unit = {
+          val it = stream.iterator()
+          def hasNext = try {
+            it.hasNext()
+          } catch {
+            case cte: ConsumerTimeoutException =>
+              props.system.log.warning("AkkaHighLevelConsumer should not see ConsumerTimeoutException")
+              false
+          }
+          props.system.log.debug("blocking on stream")
+          while (hasNext) {
+            val msg = props.msgHandler(it.next())
+            props.receiver ! msg
+          }
+        }
+      })
+
+    }
+    ()
+  }
+
+  def stop():Future[Unit] = {
```

Comment:
Probably put this whole method inside a future otherwise it's basically blocking. Similar to start